### PR TITLE
Redirect URL to Sage Intacct

### PR DESCRIPTION
**Pull request by NileshPant1999**

### CODE REVIEW

File: apps/sage_intacct/tasks.py [hunk 1]
```diff
@@ -539,7 +548,7 @@ def create_bill(expense_group: ExpenseGroup, task_log_id):
                         'Updating Attachment failed for expense group id %s / workspace id %s Error: %s',
                         expense_group.id, expense_group.workspace_id, {'error': error}
                     )
-
+
```
Comment (Sravanksk) [inline apps/sage_intacct/tasks.py:551]: rm

**Review by Sravanksk — APPROVED** [review]

File: apps/sage_intacct/tasks.py [hunk 2]
```diff
@@ -437,6 +437,11 @@ def create_expense_report(expense_group: ExpenseGroup, task_log_id):
 
             created_attachment_id = load_attachments(sage_intacct_connection, \
                                                      created_expense_report['data']['eexpenses']['RECORDNO'], expense_group)
+
+            expense_report = sage_intacct_connection.get_expense_report(created_expense_report['data']['eexpenses']['RECORDNO'], ['RECORD_URL'])
+            redirected_url_id = expense_report['eexpenses']['RECORD_URL'].split('?.r=', 1)[1]
+            created_expense_report['redirected_url_id'] = redirected_url_id
```
Comment (ashwin1111) [inline apps/sage_intacct/tasks.py:443]: Can we call it `redirect_url_id` or `url_id` or `deep_link_url`

**Review by ashwin1111 — APPROVED** [review]

File: apps/sage_intacct/tasks.py (hunk 2 above)
Comment (NileshPant1999) [reply apps/sage_intacct/tasks.py:443]: sure will do that

File: apps/tasks/views.py [hunk 3]
```diff
@@ -60,9 +60,9 @@ def get(self, request, *args, **kwargs):
         """
         Get task logs by ids
         """
-        task_logs = TaskLog.objects.filter(expense_group_id=kwargs['expense_group_id']).all()
+        task_logs = TaskLog.objects.get(expense_group_id=kwargs['expense_group_id'])
```
Comment (ashwin1111) [inline apps/tasks/views.py:63]: task_log

File: apps/sage_intacct/tasks.py [hunk 4]
```diff
@@ -625,14 +629,19 @@ def create_charge_card_transaction(expense_group: ExpenseGroup, task_log_id):
 
             sage_intacct_connection = SageIntacctConnector(sage_intacct_credentials, expense_group.workspace_id)
 
-            created_charge_card_transaction = sage_intacct_connection.post_charge_card_transaction( \
+            created_charge_card_transaction = sage_intacct_connection.post_charge_card_transaction(
                 charge_card_transaction_object, charge_card_transaction_lineitems_objects)
 
-            created_attachment_id = load_attachments(sage_intacct_connection, \
+            charge_card_transaction = sage_intacct_connection.get_charge_card_transaction(
```
Comment (ashwin1111) [inline apps/sage_intacct/tasks.py:635]: expense report changes?

Comment (NileshPant1999) [reply apps/sage_intacct/tasks.py:639]: pushed

**Review by ashwin1111 — APPROVED** [review]

**Review by Shwetabhk — APPROVED** [review]

**Pull request merged**